### DISCUSSION TITLE
How to attribute a color based on ordinal value?

### DISCUSSION THREAD

**Discussion by dsobral** [vega/vega-lite#8273] [Q&A] [answered]
Hello all,

I am generating a stacked barplot, and changing it depending on a filter. The problem is that, depending on the filter, some categories no longer appear on some of the graphs (which is fine and expected), but the colors of the same categories/values start changing between graphs.

Isn't it possible to define a scale where I colors are associated to specific category values?
eg: { 'category1': 'blue', 'category2': 'red' }, etc...?
Or at least something that guarantees that, independently of the filter, the color scale is the same between all graphs.

I'm going around in circles in the documentation and I don't seem to be able to find this...

Thanks,
Daniel

**Comment by dsobral** [comment, marked as answer]
Sorry, in the meantime I found the answer by myself.
You just have to define a domain in the scale. 
Not very obvious but simple enough...

"domain": ["category1", "category2", ...]

**Reply by domoritz** [reply]
I'm curious what you would have expected and what would be more obvious?

**Reply by dsobral** [reply]
In R I often use named vectors, where the names are categories and values are the colors. 

In a sense, something like what I suggested in my initial post:
{ 'category1': 'blue', 'category2': 'red' }

Best,
Daniel

**Reply by dsobral** [reply]
Also, when I said it was not very obvious it is because vega and vega lite seem very powerful and have many concepts, so the learning curve is a little steep if you fall outside the (excellent) available examples.
Thanks
